### PR TITLE
Implement scanner valuer via generics

### PR DESCRIPTION
**Pull request by dkorittki**
This is based on @hf-mrdachner ´s work in PR #121.
I just refactored some code, fixed a bug where nil causes a panic, and generated all Scanner/Valuer methods for all enums using code generation.
Whats missing is unit tests for testing these methods, which might be best put into tests for components and BO´s. But overall this works as intended.

PS: During code generation I found out that some package names do not follow the general structure of `enum/{{package_name}}/{{package_name}}.go` for the enums main files. I fixed this along the way.

### CODE REVIEW

**Comment by dkorittki** [comment]
Converted to Draft, need to fix builds and tests first

**Comment by dkorittki** [comment]
Fixed everything except for coverage action. This is something which needs some thought. The code I provided is auto-generated and imo should not be tested. This rather can be done on some samples inside component or BO testing. But in this Github action, the enum packages are still being coverage counted, which does not make sense. Should I exclude this directory? @hf-mrdachner

**Comment by hf-mrdachner** [comment]
> Fixed everything except for coverage action. This is something which needs some thought. The code I provided is auto-generated and imo should not be tested broadly, since its always the same. This rather can be done on some samples inside component or BO testing. But in this Github action, the enum packages are still being coverage counted, which does not make sense. Should I exclude this directory? @hf-mrdachner

yes, please

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -16,8 +16,12 @@ jobs:
...
       - name: Checkout Code
         uses: actions/checkout@v2
-      - name: Calc coverage
-        run: go test ./... -v -covermode=count -coverprofile=coverage.out
+      - name: Calc coverage BO
+        run: go test ./bo -v -covermode=count -coverprofile=coverage.bo.out
+      - name: Calc coverage COM
+        run: go test ./com -v -covermode=count -coverprofile=coverage.com.out
+      - name: Merge cover results
+        run: cat coverage.bo.out > coverage.out && cat coverage.com.out >> coverage.out
```
Comment (hf-mrdachner) [inline .github/workflows/coverage.yml:24]:
```suggestion
      - name: Calc coverage
        run: go test ./bo ./com -v -covermode=count -coverprofile=coverage.bo.out
```

**Review by hf-mrdachner — APPROVED** [review]

**Pull request merged**